compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -9077,4 +9097,45 @@ private STNode parseIntersectionTypeDescriptor(STNode leftTypeDesc, ParserRuleCo
         STNode rightTypeDesc = parseTypeDescriptor(context);
         return STNodeFactory.createIntersectionTypeDescriptorNode(leftTypeDesc, bitwiseAndToken, rightTypeDesc);
     }
+
+    /**
+     * Parse optional field access expression .
+     *
+     * @param lhsExpr Preceding expression of the optional field access
+     * @return <code>optional-field-access-expr</code>.
+     */
+    private STNode parseOptionalFieldAccessExpression(STNode lhsExpr) {
+        STNode optionalFieldAccessToken = parseOptionalFieldAccessToken();
+        STNode fieldName = parseIdentifier(ParserRuleContext.FIELD_OR_FUNC_NAME);
+        return STNodeFactory.createOptionalFieldAccessExpressionNode(lhsExpr, optionalFieldAccessToken, fieldName);
+    }
+
+    /**
+     * Parse optional-field-access-token.
+     *
+     * @return parsed node
+     */
+    private STNode parseOptionalFieldAccessToken() {
+        STToken token = peek();
+        if (token.kind == SyntaxKind.OPTIONAL_CHAINING_TOKEN) {
+            return consume();
+        } else {
+            Solution sol = recover(token, ParserRuleContext.OPTIONAL_CHAINING_TOKEN);
+            return sol.recoveredNode;
+        }
+    }
+
+    /**
+     * Parse conditional expression.
+     *
+     * @param lhsExpr Preceding expression of the question mark
+     * @return <code>conditional-expr</code>.
+     */
+    private STNode parseConditionalExpression(STNode lhsExpr) {
+        STNode questionMark = parseQuestionMark();
+        STNode middleExpr = parseExpression(OperatorPrecedence.ELVIS_CONDITIONAL, true, false);
+        STNode colon = parseColon();
+        STNode endExpr = parseExpression(OperatorPrecedence.ELVIS_CONDITIONAL, true, false);
+        return STNodeFactory.createConditionalExpressionNode(lhsExpr, questionMark, middleExpr, colon, endExpr);
+    }
```

Comment:
Assuming `qualified-identifier` gets the higher precedence `a ? b:c : d` passes as anticipated. 
But, If we consider simple `a ? b : c`, it cannot be parsed as `b:c` is parsed as qualified identifier. 
That's problem I'm facing now

Reply:
I see. May be we would have to special case that. i.e: in `expr1 ? expr2 : expr3` if colon is missing, we check whether `expr2` is a qualified identifier or not. If yes, we can split it back and treat it as the `expr2:expr3`. Not the cleanest way, but that would work i think.
WDYT?

Reply:
Well, `expr2` is not always a qualified identifier, when we have this problem. 
For example in `a ? b : c+d`, we have a binary expression. 
So depend on the situation we will have to check their children or children of children or .... for a qualified identifier. 

Sometimes there cloud be more than one qualified identifier. Splitting and re-parsing is not possible in such complex cases.

Reply:
Since we have an LL(k) parser I was thinking something like this.
When parsing a qualified identifier, we can check whether there's a spare colon ahead until lower precedence operator or end of expression token is reached. If there's no spare colon ahead, we will not parse the qualified identifier. 

We will have to use a flag to identify whether we are in a `conditional-expr` when doing the above thing. 

Another problem in this approach is we could have `mapping-constructors` when checking over the tokens ahead. So we have to make sure that we are looking for a spare colon which is not surrounded by `{`  `}`  tokens. 

WDYT?

Reply:
Ok.
I think we have to treat `a ? b : c` as an invalid code. Because according to https://github.com/ballerina-platform/ballerina-spec/issues/523, `b:c` is a var ref.

If someone want to make it working, they have to use parenthesis like:  `a ? (b) : c`.
That will simplify many things.

Reply:
Ok great. Then no trouble at all.